### PR TITLE
ENH: replace Coordinates.X,Y & Features.X,Y with Indices.X,Y

### PR DESCRIPTION
**Pull request by ambrosejcarr**
This is a longstanding issue wherein we agree that `Indices` should gain `X, Y`, `Features` should Lose `X, Y` and `Coordinates.X, .Y` and `.Z` should be used only when referring to coordinates. 

This PR makes those changes.

### CODE REVIEW

File: starfish/experiment/builder/__init__.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def build_image(
     collection = Collection()
     for fov_ix in range(fov_count):
         fov_images = TileSet(
-            [Coordinates.X, Coordinates.Y, Indices.Z, Indices.ROUND, Indices.CH],
+            [Indices.X, Indices.Y, Indices.Z, Indices.ROUND, Indices.CH],
```
Comment (dganguli) [inline starfish/experiment/builder/__init__.py]: I thought these are meant to be coordinates? @ttung ?

Comment (ttung) [reply starfish/experiment/builder/__init__.py]: I think these are supposed to be indices.

File: starfish/stack.py [hunk 2]
```diff
@@ -783,9 +783,9 @@ def write(self, filepath: str, tile_opener=None) -> None:
 
         seen_x_coords, seen_y_coords, seen_z_coords = set(), set(), set()
         for tile in self._image_partition.tiles():
-            seen_x_coords.add(tile.coordinates[Coordinates.X])
-            seen_y_coords.add(tile.coordinates[Coordinates.Y])
-            z_coords = tile.coordinates.get(Coordinates.Z, None)
+            seen_x_coords.add(tile.coordinates[Indices.X])
```
[the remark this answers is not in the document]
Comment (ttung) [reply starfish/stack.py]: I think these should be coordinates.

**Comment by ttung** [comment]
Coordinates: refers to real space.
Indices: refers to pixel space.

File: starfish/stack.py [hunk 3]
```diff
@@ -888,7 +888,7 @@ def synthetic_stack(
             tile_extras_provider = cls._default_tile_extras_provider
 
         img = TileSet(
-            {Coordinates.X, Coordinates.Y, Indices.ROUND, Indices.CH, Indices.Z},
+            {Indices.X, Indices.Y, Indices.ROUND, Indices.CH, Indices.Z},
```
[the remark this answers is not in the document]
Comment (ambrosejcarr) [reply starfish/stack.py]: This is the same as the one that tony confirmed was supposed to be Indices. I've changed the two instances of coordinates that you correctly identified were wrong. 👍

**Comment by ambrosejcarr** [comment]
Will fix this AM. I didn't check the notebooks after I rebased***. :-(

**Pull request merged**